zk/conn.go

```diff
@@ -267,7 +284,7 @@ func (c *Conn) loop() {
 
 		// Yeesh
 		if err != io.EOF && err != ErrSessionExpired && !strings.Contains(err.Error(), "use of closed network connection") {
-			log.Println(err)
+			c.logger.Printf("%s\n", err)
```

Comment:
Makes sense. I think it's fine to leave off the newline though as the logger normally treats it the same, and when using a custom logger it's nice not to have the newline.